test-avocado/checkrealms-kerberos.py

```diff
@@ -0,0 +1,121 @@
+#!/usr/bin/python
+# -*- coding: utf-8 -*-
+
+# This file is part of Cockpit.
+#
+# Copyright (C) 2015 Red Hat, Inc.
+#
+# Cockpit is free software; you can redistribute it and/or modify it
+# under the terms of the GNU Lesser General Public License as published by
+# the Free Software Foundation; either version 2.1 of the License, or
+# (at your option) any later version.
+#
+# Cockpit is distributed in the hope that it will be useful, but
+# WITHOUT ANY WARRANTY; without even the implied warranty of
+# MERCHANTABILITY or FITNESS FOR A PARTICULAR PURPOSE. See the GNU
+# Lesser General Public License for more details.
+#
+# You should have received a copy of the GNU Lesser General Public License
+# along with Cockpit; If not, see <http://www.gnu.org/licenses/>.
+
+from avocado import job
+from avocado import test
+from avocado.utils import process
+
+import os, sys, tempfile, socket, shutil, imp
+topdir = "/usr/share/avocado/tests"
+libdir = str(os.path.dirname(os.path.abspath(__file__)))+"/lib"
+sys.path.append(libdir)
+sys.path.append(topdir+"/lib")
+from testlib import *
+from libjournal import *
+
+try:
+    a=imp.load_source("", "%s/var.env" % libdir)
+    domain=a.IPADOMAIN
+    domainip=a.IPADOMAINIP
+except:
+    a=None
+    domain="cockpit.lan"
+    domainip="192.168.122.55"
+    pass
+
+domainkrb=domain.upper()
+dnamedomainipa=socket.gethostbyaddr(domainip)[0]
+dname=socket.gethostname()
+password="foobarfoo"
+
+
+KRB5_TEMPLATE = """
+[logging]
+ default = FILE:%(dir)s/krb5.log
+[libdefaults]
+ dns_lookup_realm = false
+ default_realm = %(dom)s
+ rdns = false
+[realms]
+ %(dom)s = {
+   kdc = %(addr)s
+   master_kdc = %(addr)s
+ }
+[domain_realm]
+ %(addr)s = %(dom)s
+"""
+
+JOIN_SCRIPT1 = """ curl -s --negotiate -u : https://%(namedom)s/ipa/json --header 'Referer: https://%(namedom)s/ipa' --header "Content-Type: application/json" --header "Accept: application/json" --data '{"params": [["HTTP/%(namemy)s@%(domkrb)s"], {"raw": false, "all": false, "version": "2.101", "force": true, "no_members": false}], "method": "service_add", "id": 0}' """
+JOIN_SCRIPT2 = """ ipa-getkeytab -q -s %(namedom)s -p HTTP/%(namemy)s -k /etc/krb5.keytab"""
+
+# This is here because our test framework can't run ipa VM's twice
```

Comment:
This is a out-dated comment.  checkrealms-kerberos is non-working, intermediate code, it is probably best to remove it.

With the old tests, we can only run one "ipa" flavored machine at any one time because the "ipa" flavor wants a fixed IP address.  And putting the two tests that work against the ipa machine into the same file was a trick to get them to run sequentially instead of in parallel.